doc/nyan/engine_types.md

```diff
@@ -0,0 +1,191 @@
+Engine types
+============
+
+These types are defined by the openage engine. Every nyan object in the game
+eventually inherits from one or more of these types.
+
+Normally, you'd store one or more *NyanObject*s in a `.nyan` file, but these
+are defined in the engine, so they're here only for an overview for us human beings (like me, *beep blop*).
+
+Currently, this is a **proposal**.
+This file does not perfectly (if at all) represent the already defined types in
+libopenage, but hopefully will in the future. Many classes here, if not all,
+are currently work in progress. Prepare for changes.
+
+Shoutout:
+* Everyone: Comment! Give us good ideas.
+* libopenage-devs: change/erase this. You are the ones, who (are going to)
+implement this and know what's necessary and what should be where.
+
+```
+Mod():
+	# All patches in the 'patches' member are added on load time. So every mod
+	# needs to inherit from this class and new patches to 'patches'.
+
+	name : text
+	patches : set(NyanPatch) # Add your mod's patches to this member.
+
+Tech():
+	# Every patch that needs to be evaluated after a specific event in the game
+	# (e.g. research) needs to be added to 'updates'.
+
+	name : text              # ingame name
+	description : text       # ingame description
+	updates : set(NyanPatch) # Add your new researches (or other patches) here.
+
+#Proposal: call a `Tech' `Update' in nyan, and define Tech in openage like this:
+Update():
+	updates : set(NyanPatch)
+
+HUDElement():
+	#makes an element on the left of the HUD (building or unit menu)
+	name : text
+	description : text
+	icon: text
+Tech(Update, HUDElement):
+
+
+Unit(): # WIP
```

Comment:
- Cost
- Have limit (The max number of units of this type that a player can have at an instance)
- Had limit (The max number of units of this type that a player can create)